articles/sharding_mt_app.rst

```diff
@@ -0,0 +1,172 @@
+=========================================
+
+Whether you’re building marketing analytics, a portal for e-commerce
+sites, or an application to cater to schools, if you’re building an
+application and your customer is another business then a multi-tenant
+approach is the norm. The same code runs for all customers, but each
+customer sees their own private data set, *except in some cases of
+holistic internal reporting*.
+
+Early in your application’s life customer data has a simple structure
+which evolves organically. Typically all information relates to a
+central customer/user/tenant table. With a smaller amount of data (10’s
+of GB) it’s easy to scale the application by throwing more hardware
+at it, but what happens when you’ve had enough success and data that
+you have no longer fits in memory on a single box, or you need more
+concurrency? You scale out, often painfully.
+
+This scale out model has worked well for the likes of
+`Google <http://research.google.com/pubs/pub41344.html>`__ and
+`Instagram <http://instagram-engineering.tumblr.com/post/10853187575/sharding-ids-at-instagram>`__,
+but also doesn't have to be as complicated as you might think. If
+you're able to model your multi-tenant data in the right way sharding
+can become much simpler and still give you the power you need from a
+database including joins, indexing, and more. While Citus lets you
+scale out your processing power and memory, how you model your data may
+determine the ease and flexibility you get from the system. If you're
+building a multi-tenant SaaS application hopefully the following example
+highlights how you can plan early for scaling without having to contort
+too much of your application.
+
+Tenancy
+~~~~~~~
+
+At the core of most non-consumer focused applications tenancy is already
+built in, whether you realize it or not. As we mentioned above you may
+have a users table. Let's look at a very basic SaaS schema that
+highlights this:
+
+.. code:: sql
+
+    CREATE TABLE stores (
+      id UUID,
+      owner_email VARCHAR(255),
+      owner_password VARCHAR(255),
+      name VARCHAR(255),
+      url VARCHAR(255),
+      last_login_at TIMESTAMPTZ,
+      created_at TIMESTAMPTZ
+    )
+
+    CREATE TABLE products (
+      id UUID,
+      name VARCHAR(255),
+      description TEXT,
+      price INTEGER,
+      quantity INTEGER,
+      store_id UUID,
+      created_at TIMESTAMPTZ,
+      updated_at TIMESTAMPTZ
+    )
+
+    CREATE TABLE purchases (
+      id UUID,
+      product_id UUID,
+      customer_id UUID,
+      store_id UUID,
+      price INTEGER,
+      purchased_at TIMESTAMPTZ,
+    )
+
+The above schema highlights an *overly simplified* multi-tenant
+e-commerce site. Say for example someone like an Etsy. And of course
+there are a number of queries you would run against this:
+
+List the products for a particular store:
+
+.. code:: sql
+
+    SELECT id, 
+           name,
+           price
+    FROM products
+    WHERE store_id = foo;
```

Comment:
The quoting around foo looks off.

Could we add quotes around foo here and change the quotes below to 'foo'?